File: tests/basic/variable_inline.cpp
int f(int a) {
  return a + 1;
}

int main() {
  
  // inlining of variable
  const int a = 2;
  const int b = a + a;

  int c = 3;
  int d = c;
  c = d;
  // inlining of function
  const int e = f(2);

  return 0;
}

File: tests/combi/variable_inline.ml
open Optitrust
open Target

let _ = Run.script_cpp (fun _ ->
   !! Variable.inline [cTopFunDef "test_ref"; cVarDef "b"];
   !! Variable.inline ~simpl:(fun tg -> Arith.(simpl_surrounding_expr gather (nbAny :: tg))) [cTopFunDef "test_nonconst"; cVarDef "a"];
   !! Tools.failure_expected (fun () ->
      !! Variable.inline [cTopFunDef "test_nonconst_fail"; cVarDef "a"]);

   (* more complex test *)
   !! Variable.inline [cVarDef "p"];
)
